docs/content/technical/build_process.md

```diff
@@ -0,0 +1,115 @@
+options used. Some of this is hidden from the user in `specs` files located
+[here](https://github.com/blocksds/sdk/tree/master/sys/crts). This is done
+purely for convenience. You can skip the `specs` files and copy the build
+options and definitions to your makefile, but that will force you to keep track
+of the changes in BlocksDS and to update your makefiles accordingly.
+
+The best way to understand how BlocksDS invokes `gcc` is to build a ROM that
+uses one of the default makefiles of (located
+[here](https://github.com/blocksds/sdk/tree/master/sys/default_makefiles/).
+If you build the binary with `make V=` instead of just `make`, the makefile will
+print the exact commands used at every step of the build process.
+
+TODO: Mention why custom libc is required.
+
+Note: In the past, the build process involved creating `bin` files from `elf`
+files (with `objcopy`) and using them to build the final ROM. However, `bin`
+files generated this way don't contain information about the entrypoint address,
+and it isn't possible to determine which parts are only needed for DSi.
+
+## 3. Filesystem
+
+The size of the ARM7 and ARM9 binaries is limited to the available RAM of the
+console. In order to support bigger games, the NDS ROM can contain a `NitroFS`
+filesystem with your graphics, music, etc. That way you can have a smaller CPU
+binary and load data from the filesystem whenever it's required.
+
+There is no standalone tool to generate a filesystem image. Developers are
+expected to have one or more folders with data and pass them to `ndstool` when
+the ROM is created. `ndstool` will generate a filesystem image by combining all
+the folders provided by the user.
+
+## 4. Teak DSP binaries
+
+The DSi has another general-purpose CPU that can run user code, a Teak DSP.
+Binaries for this CPU are treated as generic data. You can add them to your CPU
+binaries, or keep them in the filesystem so that they can be switched at
+runtime.
+
+## 5. NDS ROM creation
+
+`ndstool` is a program that can take the following and generate a NDS ROM out of
+it:
+
+- One ARM9 `elf` file.
+- One ARM7 `elf` file. If no file is specified, the [default
+  ARM7](https://github.com/blocksds/sdk/tree/master/sys/default_arm7) of
+  BlocksDS will be used instead.
+- One or more folders to be added to the ROM filesystem. If no folder is
+  provided, the filesystem will be kept emtpy.
+- One ROM icon (which may be animated for DSi-compatible games). This icon is
+  displayed in the NDS firmware menu. It's optional.
+
+`ndstool` reads the `elf` file of each CPU, takes every section in the `elf`
+file, and saves it to the NDS ROM as binary data. This includes `twl` sections
+as well as regular sections.
+
+It is possible to generate a NDS ROM without any DSi code or data by running
+`ndstool` with `-h 0x200`, which will force the NDS ROM header to be DS-only. In
+this case, none of the the `twl` sections will be added to the final NDS ROM.
```

Comment:
This option also generates a PassMe header; while initially intended for loading DS homebrew from Slot-2 cartridges, it is also used by some outdated homebrew application loaders.